userland/examples/tests/adc/main.c

```diff
@@ -0,0 +1,80 @@
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <tock.h>
+#include <console.h>
+#include <timer.h>
+#include <adc.h>
+
+// List of frequencies to sample at
+const uint32_t FREQS[10] =
+    {25, 100, 500, 1000, 5000, 10000, 44100, 100000, 150000, 175000};
+
+static void test_single_samples(uint8_t channel) {
+  uint16_t sample;
+
+  int err = adc_sample_sync(channel, &sample);
+  if (err < 0) {
+    printf("Error sampling ADC: %d\n", err);
+
+  } else {
+    // 12 bit, reference = VCC/2, gain = 0.5
+    // millivolts = ((sample * 2) / (2^12 - 1)) * (3.3 V / 2) * 1000
+    int millivolts = (sample * 3300) / 4095;
+    printf("ADC Reading: %d mV (raw: 0x%04x)\n", millivolts, sample);
+  }
+}
+
+static void test_sampling_buffer(uint8_t channel, int index) {
+  uint32_t length = 16;
+  uint16_t buf[length];
+  memset(buf, 0, length);
+
+  printf("%lu ADC samples at %lu Hz\n", length, FREQS[index]);
+  int err = adc_sample_buffer_sync(channel, FREQS[index], buf, length);
+  if (err < 0) {
+    printf("Error sampling ADC: %d\n", err);
+
+  } else {
+    printf("\t[ ");
+    for (uint32_t i=0; i<length; i++) {
+      // convert to millivolts
+      printf("%u ", (buf[i] * 3300) / 4095);
+    }
+    printf("]\n");
+  }
+}
+
+int main(void) {
+  putstr("[Tock] ADC Test\n");
```

Comment:
Since you are here, how would you feel about making these all consistent (putstr -> printf) in the two main.c files?